### PR TITLE
Implemented RFC0004

### PR DESCRIPTION
**Pull request by pksunkara**
This PR implements the changes proposed in RFC0004.

This PR depends refractproject/rfcs#10 to be merged first.

### CODE REVIEW

File: namespaces/api-description-namespace.md [hunk 1]
```diff
@@ -177,13 +179,17 @@ Note: At the moment only the HTTP protocol is supported.
         Definition of any input message-body attribute for this transition.
 
     - `contentTypes` (array[String]) - A collection of content types that MAY be used for the transition.
-- `content` (array[HTTP Transaction]) - Array of transaction examples.
+- `content` (array)
+    - (Copy) - Textual information of the transition.
+    - (HTTP Transaction) An instance of transaction example.
+
+        The `content` MAY include multiple `HTTP Transaction` elements.
```
Comment (smizell) [inline namespaces/api-description-namespace.md]: I'm assuming this means the content of the Transition element.

Comment (smizell) [inline namespaces/api-description-namespace.md]: Might be good to clarify that.

File: namespaces/api-description-namespace.md [hunk 2]
```diff
@@ -177,13 +179,17 @@ Note: At the moment only the HTTP protocol is supported.
...
 
-    Transaction examples are protocol-specific examples of REST transaction
-    that was initialized by exercising a transition.
+        Transaction examples are protocol-specific examples of REST transaction
```
Comment (smizell) [inline namespaces/api-description-namespace.md]: May need to say "of a REST transaction"

File: namespaces/api-description-namespace.md (hunk 1 above)
Comment (pksunkara) [inline namespaces/api-description-namespace.md]: Well, the same sentence is getting used everywhere else in the document. Ex: https://github.com/refractproject/refract-spec/pull/41/files#diff-47715162e9c85197ac4ed245268438e6R102

File: namespaces/api-description-namespace.md [hunk 3]
```diff
@@ -345,14 +350,16 @@ message pair. A transaction example MUST contain exactly one HTTP request and on
 #### Properties
 
 - `element`: httpTransaction (string, fixed)
-- `content` (array, fixed) - Request response message pair (tuple).
+- `content` (array) - Request response message pair (tuple).
```
Comment (smizell) [inline namespaces/api-description-namespace.md]: "Request and response message pair (tuple)"

Just to be clear

**Comment by smizell** [comment]
@pksunkara Looks good to me. One thing I'm wondering about is, do you want to say multiple copy elements may be used in the places in which they are found?

**Comment by pksunkara** [comment]
I think that is a given since we did not add `fixed` or didn't explicitly state any restrictions.

**Comment by smizell** [comment]
> I think that is a given since we did not add fixed or didn't explicitly state any restrictions.

Sounds good. Will keep this guideline in mind in the future.

File: namespaces/api-description-namespace.md (hunk 1 above)
Comment (smizell) [inline namespaces/api-description-namespace.md]: Will leave up to you.

**Comment by pksunkara** [comment]
Updated.

**Comment by smizell** [comment]
@pksunkara Just one last question, other than that, :+1:

**Comment by smizell** [comment]
:+1: Once RFC is good to go

File: namespaces/api-description-namespace.md [hunk 4]
```diff
@@ -95,6 +96,7 @@ The Resource representation with its available transitions and its data.
     - `href` (Templated Href) - URI Template of this resource.
     - `hrefVariables` (Href Variables) - Definition of URI Template variables used in the `href` property.
 - `content` (array)
+    - (Copy) - Textual information of the resource.
```
Comment (zdne) [inline namespaces/api-description-namespace.md]: "Textual information of the resource." does not sound right. What about something like "resource description's copy text" ?

File: namespaces/api-description-namespace.md [hunk 5]
```diff
@@ -177,13 +179,17 @@ Note: At the moment only the HTTP protocol is supported.
         Definition of any input message-body attribute for this transition.
 
     - `contentTypes` (array[String]) - A collection of content types that MAY be used for the transition.
-- `content` (array[HTTP Transaction]) - Array of transaction examples.
+- `content` (array)
+    - (Copy) - Textual information of the transition.
```
Comment (zdne) [inline namespaces/api-description-namespace.md]: -ditto-

File: namespaces/api-description-namespace.md [hunk 6]
```diff
@@ -177,13 +179,17 @@ Note: At the moment only the HTTP protocol is supported.
...
+    - (HTTP Transaction) - An instance of transaction example.
+
+        The `content` MAY include multiple `HTTP Transaction` elements.
```
Comment (zdne) [inline namespaces/api-description-namespace.md]: this is also the case for Copy, not sure if we need to say it as it should be clear from the MSON notation

File: namespaces/api-description-namespace.md [hunk 7]
```diff
@@ -345,13 +350,15 @@ message pair. A transaction example MUST contain exactly one HTTP request and on
 #### Properties
 
 - `element`: httpTransaction (string, fixed)
-- `content` (array, fixed) - Request response message pair (tuple).
+- `content` (array) - Request and response message pair (tuple).
+    - (Copy) - Textual information of the HTTP Transaction.
```
Comment (zdne) [inline namespaces/api-description-namespace.md]: see my comment on " Textual information of the.." above

File: namespaces/api-description-namespace.md [hunk 8]
```diff
@@ -437,6 +441,7 @@ of Data structure or assets.
 - `attributes`
     - `headers` (HTTP Headers)
 - `content` (array)
+    - (Copy) - Textual information of the payload.
```
Comment (zdne) [inline namespaces/api-description-namespace.md]: -ditto-

**Comment by pksunkara** [comment]
Updated.

**Comment by danielgtaylor** [comment]
:+1: looks good, just waiting on the RFC. Please don't forget to update the [changelog](https://github.com/refractproject/refract-spec/blob/master/CHANGELOG.md), possibly update the spec version & create a new tag if needed.

**Comment by smizell** [comment]
@pksunkara This is ready to go, but as @danielgtaylor said, I think it would be good to update changelog and bump spec version number. Once done, I'll merge.

**Comment by pksunkara** [comment]
Bumped version and updated changelog.

**Comment by smizell** [comment]
@pksunkara So, first, please forgive me. I think we may have something off here, as when I think about it, the version is mostly for the spec itself, not the namespaces. In looking at this, I am wondering if we need this version bump and changelog. What do you think?

**Comment by pksunkara** [comment]
What is Refract spec? Is it the definition of refract (`element`, `attributes`, `content`)? Or is it the above collection of namespaces? Should we maintain separate versions for namespaces and refract? We need to decide this later.

For now, let's just bump the only version available.

**Comment by smizell** [comment]
@pksunkara Merging. In this repo, there is a Refract spec and then a folder with namespace files. We may need separate versions for these namespaces unless we want to manage them as one big spec. Seems like having separate versions is the correct route.

**Comment by danielgtaylor** [comment]
@smizell :+1: to separate versions, otherwise we are going to run into trouble.

**Pull request merged**